KarhooSDK/Service/Loyalty/LoyaltyBurn/KarhooLoyaltyBurnInteractor.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  KarhooLoyaltyBurnInteractor.swift
+//  KarhooSDK
+//
+//  Created by Edward Wilkins on 17/11/2021.
+//  Copyright © 2021 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+
+final class KarhooLoyaltyBurnInteractor: LoyaltyBurnInteractor {
+    
+    private let requestSender: RequestSender
+    private var identifier: String?
+    private var currency: String?
+    private var amount: Int?
+    
+    init(requestSender: RequestSender = KarhooRequestSender(httpClient: TokenRefreshingHttpClient.shared)) {
+        self.requestSender = requestSender
+    }
+    
+    func set(identifier: String, currency: String, amount: Int) {
+        self.identifier = identifier
+        self.currency = currency
+        self.amount = amount
+    }
+    
+    func execute<T>(callback: @escaping CallbackClosure<T>) where T : KarhooCodableModel {
+        requestSender.requestAndDecode(payload: nil,
```

Comment:
I don't think if identifier or currency are missing the API would work. We need to implement a `guard let ` logic to avoid any issues.

Reply:
Shouldn't need to do anything for the identifier, but can add a guard let for the currency at least. The identifier is pretty much a global thing in the SDK and it knows that if no id is passed through, it will/should error by default.

Reply:
Same as for the earn interactor guard statement. We need to return an error via the callback before returning.